dkc/core/models/folder.py

```diff
@@ -82,6 +84,25 @@ def ancestors(self) -> models.query.RawQuerySet[Folder]:
             [self.pk, Folder.MAX_TREE_HEIGHT + 1],
         )
 
+    @transaction.atomic
```

Comment:
We should consider whether a read-committed txn isolation level is sufficient to avoid race conditions here. I know we don't yet support folder move, but with the understanding that we might someday, I think it's possible that there's a race condition if a folder was moved while one of its descendants was being uploaded into. I believe a repeatable-read isolation level would fix that.

Reply:
After thinking about this more, I actually think read committed may be sufficient. At least, I'm fine with merging this as-is, until we see what a move implementation would look like in terms of queries.

Reply:
@zachmullen I had the same concerns, and reached the same conclusion about waiting.